cognite/neat/rules/models/_rules/_types.py

```diff
@@ -211,3 +226,167 @@ def _check_parent(value: list[ParentClass]) -> list[ParentClass]:
         ),
     ),
 ]
+
+# Sentinel value. This is used to indicate that no prefix is set for an entity.
+Undefined = type(object())
+
+
+# mypy does not like the sentinel value, and it is not possible to ignore only the line with it below.
+# so we ignore all errors beyond this point.
+# mypy: ignore-errors
+@total_ordering
+class Entity(BaseModel, arbitrary_types_allowed=True):
```

Comment:
You moved this class from:
https://github.com/cognitedata/neat/blob/7c2e187803784889cb33ceffe938f25930fc1efe/cognite/neat/rules/models/_base.py#L54

Reply:
Yes, to avoid changing anything to the existing.